pkg/gardenlet/controller/federatedseed/seed_control.go

```diff
@@ -134,6 +130,17 @@ func (c *Controller) seedDelete(obj interface{}) {
 
 // Run starts the FederatedSeed Controller
 func (c *Controller) Run(ctx context.Context, workers int) {
+	c.seedInformer.AddEventHandler(cache.FilteringResourceEventHandler{
+		FilterFunc: controllerutils.SeedFilterFunc(confighelper.SeedNameFromSeedConfig(c.config.SeedConfig), c.config.SeedSelector),
+		Handler: cache.ResourceEventHandlerFuncs{
+			AddFunc:    c.seedAdd,
+			UpdateFunc: c.seedUpdate,
+			DeleteFunc: c.seedDelete,
+		},
+	})
+
+	c.seedHasSynced = c.seedInformer.HasSynced
+
 	var waitGroup sync.WaitGroup
 
 	if !cache.WaitForCacheSync(ctx.Done(), c.seedHasSynced) {
```

Comment:
Can you remove the `seedHasSynced` member?
```suggestion
	var waitGroup sync.WaitGroup

	if !cache.WaitForCacheSync(ctx.Done(), c.seedInformer.HasSynced) {
```